### PR TITLE
[Transform] improve transform log/audit correctness when stopped

### PR DESCRIPTION
**Pull request by hendrikmuhs**
move auditor and logging for transform being stopped, so it gets reported after
the p-task has been marked for completion

The change is motivated by a bug investigation where we saw that logging/audit reported a stopped transform although the transform was still running. Something between the `onStop` call and the p-task removal seemed to got stuck. I still don't know why, but this change fixes the wrong logging/audit. FWIW `afterFinishOrFailure` wasn't there when logging got added at the original place.

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java [hunk 1]
```diff
@@ -646,12 +649,6 @@ protected void onFailure(Exception exc) {
         }
     }
 
-    @Override
-    protected void onStop() {
-        auditor.info(transformConfig.getId(), "Transform has stopped.");
-        logger.info("[{}] transform has stopped.", transformConfig.getId());
```
Comment (droberts195) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java:652]: Maybe it's still worth logging _something_ here so we can detect the situation of something hanging in between this and `afterFinishOrFailure`?

I agree we shouldn't audit two notifications, but maybe just a log message could be helpful and it shouldn't be _that_ noisy.

**Review by droberts195 — APPROVED** [review]
LGTM

I left one suggestion but happy to merge with or without that.

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java (hunk 1 above)
Comment (hendrikmuhs) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java:652]: :+1: I added the log back as `debug`.

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Comment by droberts195** [comment]
It's been observed that `afterFinishOrFailure` gets called multiple times per stop. As a result this change causes multiple audit messages, so has been reverted until we understand why `afterFinishOrFailure` gets called multiple times.

**Pull request merged**